crypto/init.c

```diff
@@ -243,17 +243,12 @@ DEFINE_RUN_ONCE_STATIC(ossl_init_no_add_algs)
 
 static CRYPTO_ONCE config = CRYPTO_ONCE_STATIC_INIT;
 static int config_inited = 0;
-static const char *appname;
+static const OPENSSL_INIT_SETTINGS *conf_settings = NULL;
 DEFINE_RUN_ONCE_STATIC(ossl_init_config)
 {
-#ifdef OPENSSL_INIT_DEBUG
-    fprintf(stderr,
-            "OPENSSL_INIT: ossl_init_config: openssl_config(%s)\n",
-            appname == NULL ? "NULL" : appname);
-#endif
-    openssl_config_int(appname);
+    int ret = openssl_config_int(conf_settings);
     config_inited = 1;
```

Comment:
hmm, if this function can return a fault, then it should not set config_inited to 1, or what do you think?

Reply:
I still think it should run just once.  I don't think there's a good way to recover. The application can either choose to run with partial configuration, or bail out.  I am willing to listen to reason.

Note that previously we just ran regardless, so the application can still do that.  If the application has the luxury of rejecting/reporting bad configs and not running, it gets to decide.

Reply:
Yes, we don't presently do anything with that, it just set and never used. A good optimizing compiler might eliminate it entirely.